ecp5/main.cc

```diff
@@ -59,6 +59,8 @@ po::options_description ECP5CommandHandler::getArchOptions()
     specific.add_options()("um5g-45k", "set device type to LFE5UM5G-45F");
     specific.add_options()("um5g-85k", "set device type to LFE5UM5G-85F");
 
+    specific.add_options()("sa-placer", "use pure simulated annealing placer instead of HeAP analytic placer");
```

Comment:
So ecp5 has a new `"sa-placer"` option, whereas ice40 has a new `"heap-placer"` option. Intentional?

Reply:
SA remains the default for iCE40, which is a more stable architecture and small enough for speed to be less of a problem for most people. SA tends to win on iCE40 QoR too.

HeAP becomes the default for ECP5, which has major runtime (into several hours) issues with SA, but has mostly experimental users. HeAP tens to win for ECP5 QoR because it is better at moving chains which the ECP5 has lots of due to distributed RAM and cascade muxes. A SA fallback is still available and will be used automatically if HeAP is disabled in CMake (e.g. because Eigen3 isn't easily available for the target).

Reply:
Makes sense. Should write that down somewhere. I would encourage you to fixate on one argument string, and then set it to default to one value for ice40, and another value for ecp5 though...